test/cloudtest/test_recovery.py

```diff
@@ -0,0 +1,252 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+import subprocess
+from dataclasses import dataclass
+from textwrap import dedent
+
+import pytest
+
+from materialize.cloudtest.app.materialize_application import MaterializeApplication
+
+
+@dataclass
+class ClusterSetup:
+    testcase_index: int
+    storage_replication_factor: int = 1
+    compute_replication_factor: int = 1
+
+    def get_storage_name(self) -> str:
+        return f"storage_t{self.testcase_index}"
+
+    def get_compute_name(self) -> str:
+        return f"compute_t{self.testcase_index}"
+
+    def get_counter_source_name(self) -> str:
+        return f"counter_{self.testcase_index}"
+
+    def get_mv_name(self) -> str:
+        return f"mv_{self.testcase_index}"
+
+
+def populate(mz: MaterializeApplication, cluster_setup: ClusterSetup) -> None:
```

Comment:
Consider dropping all the clusters here , as well as the `materialize` database . Then you can recreate them all using the same name and you do not have to track `testcase_index` at all. 

That other cloudtest was giving each sub-test a separate index because it had to create a separate kafka topic to feed it (deletion of kafka topics is prone to race conditions, so kafka topic names can not be easily reused). However, in your case, you control everything inside Mz, so you can drop all the objects from the previous test at will.

Reply:
Good idea!

✅